application/cache/cache.go

```diff
@@ -0,0 +1,233 @@
+		if cachedPrivately && isCompleted(request) {
+			delete(requestsMap, request.UUID.String())
+			return CacheWrite(ctx, organization.Name, requestsMap)
+		}
+		// was cached privately, but request visbility has changed to public.
+		// Need to remove from private and add to public cache.
+		if cachedPrivately && isPublic(request) {
+			delete(requestsMap, request.UUID.String())
+			if err := CacheWrite(ctx, organization.Name, requestsMap); err != nil {
+				return err
+			}
+		}
+
+		// should be cached privately. Need to update or create an entry in the private cache.
+		// Cannot return yet in case the request was previously public (must remove from public cache)
+		if isPrivate(request) {
+			requestAbridged, err := conversions.ConvertRequestToAPITypeAbridged(ctx, request)
+			if err != nil {
+				return err
+			}
+			requestsMap[request.UUID.String()] = requestAbridged
+			if err := CacheWrite(ctx, organization.Name, requestsMap); err != nil {
+				return err
+			}
+		}
+
+		// if request was and remains public and active, we update it below
+
+	}
+	// update public cache
+	newEntryCreated, requestsMap, err = getOrCreateCacheEntryPublic(ctx)
+	if err != nil {
+		return err
+	}
+	// may need to update existing cache entry with new data
+	if !newEntryCreated {
+		_, cachedPublicly := requestsMap[request.UUID.String()]
+		// was cached publicly, but request visbility has completed. Just need to remove from public cache.
+		if cachedPublicly && isCompleted(request) {
+			delete(requestsMap, request.UUID.String())
+			return CacheWrite(ctx, organization.Name, requestsMap)
+		}
+		// was cached publicly, but request visbility has changed to private.
+		// Already added to private cache, now need to add to public cache.
+		if cachedPublicly && isPrivate(request) {
+			delete(requestsMap, request.UUID.String())
+			if err := CacheWrite(ctx, organization.Name, requestsMap); err != nil {
+				return err
+			}
+		}
+
+		// should be cached publicly. Just need to update or create an entry in the public cache,
+		// as we've already dealt with the case where visibility was originally private
+		if isPublic(request) {
+			requestAbridged, err := conversions.ConvertRequestToAPITypeAbridged(ctx, request)
+			if err != nil {
+				return err
+			}
+			requestsMap[request.UUID.String()] = requestAbridged
+			return CacheWrite(ctx, organization.Name, requestsMap)
+		}
+	}
+	return nil
+}
+
+// Gets cache entry for organization, or creates one if none exists
+// Returns true if new cache entry was created
+func getOrCreateCacheEntryPrivate(ctx context.Context, organization models.Organization) (bool, map[string]api.RequestAbridged, error) {
+	var requestsMap map[string]api.RequestAbridged
+	// if a cache value does not exist, we create it
+	if err := CacheRead(ctx, organization.Name, requestsMap); err != nil {
+		tx := models.Tx(ctx)
+
+		var org models.Organization
+		if err := org.FindByUUID(tx, organization.UUID.String()); err != nil {
```

Comment:
If the `organization` variable has its `id` but just needs to be refetched from the DB, you can use `tx.Reload(&organization)`. Or, maybe you don't need to refresh it at all. Not sure what's needed here.

Reply:
I don't actually need to refresh here, as the caller passes in a valid organization. (This was left over from some earlier logic where I was using an api.Request rather than a models.Request and thus didn't have the organization UUID available, and I forgot to take this out.)